### PR TITLE
[#1782] Add max position params to speech recognition

### LINKED ISSUE #1782: Speech recognition can OOM with large audio sequences
## 🚀 Feature Request

Unlike other tasks such as `translation`, `speech_recognition` doesn't support the possibility to define max sequence lengths in order to filter examples not respecting them. This can lead to OOM during the training. So it would be nice to have the same feature in this task too.

### Motivation

Without it, several OOM can happen. This can cause other examples to be ignored and a slower training.

### Pitch

We propose to add the `--max-source-positions` and `--max-target-positions` args to the `speech_recognition` task.

### Alternatives

The only alternative I can think of is ignoring the OOMs...

### PR DESCRIPTION
**Pull request by mgaido91**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/pytorch/fairseq/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  

## What does this PR do?
Fixes #1782.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by mgaido91** [comment]
the failure seems unrelated to this PR, but I don't know how to retrigger the tests, can anyone help me? Thanks.

**Comment by erip** [comment]
@mgaido91 Windows never passes.

File: examples/speech_recognition/tasks/speech_recognition.py [hunk 1]
```diff
@@ -77,8 +78,8 @@ def add_args(parser):
         parser.add_argument(
             "--silence-token", default="\u2581", help="token for silence (used by w2l)"
         )
-        parser.add_argument('--max-source-positions', default=2048, type=int, metavar='N',
-                            help='max number of tokens in the source sequence')
+        parser.add_argument('--max-source-positions', default=sys.maxsize, type=int, metavar='N',
+                            help='max number of frames in the source sequence')
         parser.add_argument('--max-target-positions', default=1024, type=int, metavar='N',
```
Comment (okhonko) [inline examples/speech_recognition/tasks/speech_recognition.py]: We can make this default value as `sys.maxsize` as well, unless we know that there are no utterances in librispeech dataset with targets BPE units longer than 1024 .

Comment (mgaido91) [reply examples/speech_recognition/tasks/speech_recognition.py]: Thanks for your reviews! I guess it depends on the BPE size used. In my case, I am using 8000 BPE and the longest target contains 132 tokens. 1024 is the default max for the translation task too. I can change it to `sys.maxsize` if you prefer, but IMHO this is reasonable. Thanks.

Comment (okhonko) [reply examples/speech_recognition/tasks/speech_recognition.py]: Thanks for checking!

**Comment by mgaido91** [comment]
Thanks for the review @okhonko . Is there anything else I can do to push this PR forward? Thanks in advance for the guidance.

**Pull request closed without merging**